src/pkg/message/credentials.go

```diff
@@ -0,0 +1,163 @@
+package message
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/defenseunicorns/zarf/src/config"
+	"github.com/defenseunicorns/zarf/src/types"
+	"github.com/pterm/pterm"
+)
+
+// Common constants for printing credentials
+const (
+	loggingUsername = "zarf-admin"
+
+	RegistryKey     = "registry"
+	RegistryReadKey = "registry-readonly"
+	GitKey          = "git"
+	GitReadKey      = "git-readonly"
+	ArtifactKey     = "artifact"
+	LoggingKey      = "logging"
+)
+
+// PrintCredentialTable displays credentials in a table
+func PrintCredentialTable(state types.ZarfState, componentsToDeploy []types.DeployedComponent) {
+	if len(componentsToDeploy) == 0 {
+		componentsToDeploy = []types.DeployedComponent{{Name: "logging"}, {Name: "git-server"}}
+	}
+
+	// Set output to os.Stderr to avoid creds being printed in logs
+	pterm.SetDefaultOutput(os.Stderr)
+
+	pterm.Println()
+	loginTableHeader := pterm.TableData{
+		{"     Application", "Username", "Password", "Connect", "Get-Creds Key"},
+	}
+
+	loginTable := pterm.TableData{}
+	if state.RegistryInfo.InternalRegistry {
+		loginTable = append(loginTable, pterm.TableData{
+			{"     Registry", state.RegistryInfo.PushUsername, state.RegistryInfo.PushPassword, "zarf connect registry", RegistryKey},
+			{"     Registry (read-only)", state.RegistryInfo.PullUsername, state.RegistryInfo.PullPassword, "zarf connect registry", RegistryReadKey},
+		}...)
+	}
+
+	for _, component := range componentsToDeploy {
+		// Show message if including logging stack
+		if component.Name == "logging" {
+			loginTable = append(loginTable, pterm.TableData{{"     Logging", loggingUsername, state.LoggingSecret, "zarf connect logging", LoggingKey}}...)
+		}
+		// Show message if including git-server
+		if component.Name == "git-server" {
+			loginTable = append(loginTable, pterm.TableData{
+				{"     Git", state.GitServer.PushUsername, state.GitServer.PushPassword, "zarf connect git", GitKey},
+				{"     Git (read-only)", state.GitServer.PullUsername, state.GitServer.PullPassword, "zarf connect git", GitReadKey},
+				{"     Artifact Token", state.ArtifactServer.PushUsername, state.ArtifactServer.PushToken, "zarf connect git", ArtifactKey},
+			}...)
+		}
+	}
+
+	if len(loginTable) > 0 {
+		loginTable = append(loginTableHeader, loginTable...)
+		_ = pterm.DefaultTable.WithHasHeader().WithData(loginTable).Render()
+	}
+
+	// Restore the log file if it was specified
+	if !config.SkipLogFile {
+		UseLogFile()
+	}
+}
+
+// PrintComponentCredential displays credentials for a single component
+func PrintComponentCredential(state types.ZarfState, componentName string) {
+	switch strings.ToLower(componentName) {
+	case LoggingKey:
+		Notef("Logging credentials (username: %s):", loggingUsername)
+		fmt.Println(state.LoggingSecret)
+	case GitKey:
+		Notef("Git Server push password (username: %s):", state.GitServer.PushUsername)
+		fmt.Println(state.GitServer.PushPassword)
```

Comment:
```suggestion
Notef("Git Server push password (username: %s):", state.GitServer.PushUsername)
```

Reply:
i realize this is a preliminary PR and this may be just for debugging purposes so feel free to disregard

Reply:
These are intentional - they are printing the secrets to standard out for those commands to be ingested into other things (i.e.)

```
GIT_PASSWORD=$(zarf tools get-creds git) curl -u zarf-git-user:${GIT_PASSWORD} http://localhost:45362/api/v1/user
```